### PR TITLE
include: migrate to SPDX identifier

### PR DESCRIPTION
**Pull request by jerpelea**
## Summary
Most tools used for compliance and SBOM generation use SPDX identifiers This change brings us a step closer to an easy SBOM generation.

ignored for now 
include/search.h
include/crypto/cast.h 
include/crypto/cryptodev.h 
include/crypto/cryptosoft.h 
include/crypto/md5.h 
include/crypto/poly1305.h 
include/crypto/rijndael.h 
include/crypto/sha1.h 
include/crypto/xform.h 
include/nuttx/input/x11_keysym.h
include/nuttx/input/x11_keysymdef.h
include/nuttx/input/x11_xf86keysym.h
include/nuttx/lcd/ili9488.h 

## Impact
SPDX

## Testing
CI

### CODE REVIEW

**Review by cederom — APPROVED** [review]
Thank you @jerpelea !! :-)

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



The PR summary is missing crucial information:

* **Why is this change necessary?**  Clearly state the problem this PR solves. Is SPDX support missing entirely? Are current methods insufficient?
* **What functional part of the code is being changed?** Be specific.  "SPDX" is too broad.  List affected files/modules.
* **How does the change exactly work?**  Does it add new APIs? Modify existing ones? Implement a specification?

**Impact Assessment:**

* **Is new feature added? Is existing feature changed?**  Needs specific YES/NO answers.
* **Impact on user?**  Will they need new tools, configurations, etc.?
* **Impact on build?**  Does it add dependencies, change build flags?
* **All other impact points:**  Similarly need specific YES/NO + explanation.

**Testing:**

* **"CI" is insufficient.**  List the specific architectures/boards tested *locally* to validate the changes.
* **No "before" logs.**  How do we know what behavior changed without a baseline?

**Concisely:**  The PR does **NOT** meet NuttX requirements in its current state. It lacks detail and evidence of thorough testing.

**Review by acassis — APPROVED** [review]

File: include/nuttx/net/netdev.h [hunk 1]
```diff
@@ -1,16 +1,10 @@
...
- *   Copyright (C) 2007, 2009, 2011-2018 Gregory Nutt. All rights reserved.
- *   Author: Gregory Nutt <gnutt@nuttx.org>
- *
- * Derived largely from portions of uIP with has a similar BSD-styple
- * license:
- *
- *   Copyright (c) 2001-2003, Adam Dunkels.
- *   All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SPDX-FileCopyrightText: 2007, 2009, 2011-2018 Gregory Nutt. All rights reserved.
```
Comment (wengzhe) [inline include/nuttx/net/netdev.h:5]: Hi @jerpelea , I found that our `checkpatch.sh`(`nxstyle.c`) does not allow this long line, but I don't know which one should we modify, this header file or `nxstyle.c`?
```bash
./tools/checkpatch.sh -f include/nuttx/net/netdev.h
include/nuttx/net/netdev.h:5:83: error: Long line found
```

Comment (jerpelea) [reply include/nuttx/net/netdev.h:5]: it is a long line nxtools is right

**Pull request merged**